src/toil/jobStores/aws/jobStore.py

```diff
@@ -74,38 +82,70 @@ def copyKeyMultipart(srcKey, dstBucketName, dstKeyName, headers=None):
     :rtype: boto.s3.multipart.CompletedMultiPartUpload
     :return: An object representing the completed upload.
     """
+    def copyPart(partIndex):
+        if exceptions:
+            return None
+        else:
+            try:
+                for attempt in retry_s3():
+                    with attempt:
+                        # Note that part_num is indexed from 1 while start and end are indexed from
+                        # 0. Notice also that the partIndex variable is indexed from 0.
+                        part = upload.copy_part_from_key(src_bucket_name=srcKey.bucket.name,
+                                                         src_key_name=srcKey.name,
+                                                         src_version_id=srcKey.version_id,
+                                                         part_num=partIndex + 1,
+                                                         start=partIndex * partSize,
+                                                         end=min(partIndex * partSize + partSize,
+                                                                 totalSize) - 1,
+                                                         headers=headers)
+                log.debug('Successfully copied part %d of %d' % (partIndex, totalParts))
```

Comment:
https://github.com/BD2KGenomics/toil/wiki/Python-Dos-and-Don'ts#dont-use-format-or--to-format-log-messages